plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -366,7 +365,7 @@ else if (column.getBaseHiveColumnIndex() < fileColumns.size()) {
                 Type readType = column.getType();
                 if (orcColumn != null) {
                     int sourceIndex = fileReadColumns.size();
-                    Optional<TypeCoercer<?, ?>> coercer = createCoercer(orcColumn.getColumnType(), readType, getTimestampPrecision(session));
```

Comment:
Just a question: We are no longer relay on a session?

Reply:
Previously we rely on the session to determine the timestamp precision so that we could read the timestamp column with that precision and now we would read them as `NANOSECOND` irrespective of the precision configured.